shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm

```diff
@@ -288,7 +288,8 @@ + (NSString*)domainNetworkPolicy:(NSDictionary*)appTransportSecurity {
   NSData* jsonData = [NSJSONSerialization dataWithJSONObject:networkConfigArray
                                                      options:0
                                                        error:NULL];
-  return [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+  [networkConfigArray release];
```

Comment:
`networkConfigArray` was leaked.

Reply:
The style in projects I worked on prior to the advent of ARC was that unless there was a compelling reason not to--e.g., code that's called in a tight loop--it's far better to `autorelease` as part of the `alloc/init` statement, because then there's no chance of introducing a leak later via early return. It also made it much easier to review code for correctness, because any line that did an `alloc/init` but not an `autorelease` was immediately suspect, triggering a check for a corresponding `release`.

So tl;dr, I would make this part of line 276 instead.